### ISSUE TITLE
Swap Loss not Decreasing in Contrastive Learning 

### ISSUE THREAD

**Issue by P1terQ** [OpenRobotLab/HIMLoco#11]
In the paper,  the authors highlight that using contrastive learning can carry more precious environmental information, and the t-SNE visualization indicates the stronger representation ability of this method. 

However, when I attempted to run the open-source code, the swap loss didn't drop, regardless of short or long iterations.

I am wondering if this is an expected behavior for contrastive learning algorithms, or if there might be an issue with my implementation.
![image](https://github.com/user-attachments/assets/91677396-5356-42ef-af91-06f98cfaf672)
![image](https://github.com/user-attachments/assets/986870ef-25e0-4489-8058-0ffd6ccc6648)

**Comment by kc-ustc** [comment]
Thanks for your sharing! I have met the same problms about _swap loss_.And I also try ablation study without contrastive learning by setting swap loss = 0,finding that the linear velocity tracking reward during training seems to make no difference with original version.

**Comment by mahanxin** [comment]
@Junfeng-Long
I  have met the same proplem about swap loss, the curve is as the following figure

![Image](https://github.com/user-attachments/assets/96007d26-dcb6-4892-8be2-cbae33c15888)
and I also conduct t-SNE to analyze the latent space output, but unfortunately it could not identify the terrain.

**Comment by CSU-BossYu** [comment]
I have encountered exactly the same issue, and I agree with your view that the teacher and student encoders in contrastive learning take shortcuts instead of learning terrain features.
